Content.Client/Singularity/Visualizers/RadiationCollectorSystem.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using Content.Shared.Singularity.Components;
+using Robust.Client.Animations;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Singularity.Visualizers;
+
+public sealed class RadiationCollectorSystem : VisualizerSystem<RadiationCollectorComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RadiationCollectorComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<RadiationCollectorComponent, AnimationCompletedEvent>(OnAnimationCompleted);
+    }
+
+    private void OnComponentInit(EntityUid uid, RadiationCollectorComponent comp, ComponentInit args)
+    {
+        comp.ActivateAnimation = new Animation {
+            Length = TimeSpan.FromSeconds(0.8f),
+            AnimationTracks = {
+                new AnimationTrackSpriteFlick() {
+                    LayerKey = RadiationCollectorVisualLayers.Main,
+                    KeyFrames = {new AnimationTrackSpriteFlick.KeyFrame(comp.ActivatingState, 0f)}
+                }, // TODO: Make this play a sound when activating a radiation collector.
+            }
+        };
+
+        comp.DeactiveAnimation = new Animation {
+            Length = TimeSpan.FromSeconds(0.8f),
+            AnimationTracks = {
+                new AnimationTrackSpriteFlick() {
+                    LayerKey = RadiationCollectorVisualLayers.Main,
+                    KeyFrames = {new AnimationTrackSpriteFlick.KeyFrame(comp.DeactivatingState, 0f)}
+                }, // TODO: Make this play a sound when deactivating a radiation collector.
+            }
+        };
+    }
+
+    private void OnAnimationCompleted(EntityUid uid, RadiationCollectorComponent comp, AnimationCompletedEvent args)
+    {
+        if (args.Key != RadiationCollectorComponent.AnimationKey
+        ||  !AppearanceSystem.TryGetData<RadiationCollectorVisualState>(uid, RadiationCollectorVisuals.VisualState, out var state))
+            return;
+
+        switch(state)
+        {
+            case RadiationCollectorVisualState.Activating:
+                AppearanceSystem.SetData(uid, RadiationCollectorVisuals.VisualState, RadiationCollectorVisualState.Active);
+                break;
+            case RadiationCollectorVisualState.Deactivating:
+                AppearanceSystem.SetData(uid, RadiationCollectorVisuals.VisualState, RadiationCollectorVisualState.Deactive);
```

Comment:
just make an UpdateAppearance method that gets called here and in OnAppearanceChange